server/src/main/java/org/elasticsearch/common/xcontent/support/XContentMapValues.java

```diff
@@ -191,62 +192,148 @@ public static Object extractValue(String path, Map<?, ?> map, Object nullValue)
     }
 
     private static Object extractValue(String[] pathElements, int index, Object currentValue, Object nullValue) {
```

Comment:
Can we separate the two functions for now? Extracting and replacing a value should be considered different and I'd prefer that we don't introduce changes in this hot path for search requests.